tests/lib/rule-tester/rule-tester.js

```diff
@@ -2376,24 +2335,37 @@ describe("RuleTester", () => {
                 }
             };
 
-            ruleTester.run("rule-with-null-schema", ruleWithNullSchema, {
+            assert.throws(() => ruleTester.run("rule-with-null-schema", ruleWithNullSchema, {
                 valid: [],
                 invalid: [
                     { code: "var foo = bar;", options: [{ foo: true }], errors: 1 }
                 ]
-            });
+            }),
+            "\"rule-with-null-schema\" rule has options but is missing the \"meta.schema\" property. Please add a schema: https://eslint.org/docs/developer-guide/working-with-rules#options-schemas");
+        });
+
+        it("throws an error when using no-op schema {}", () => {
+            const ruleWithEmptyObjectSchema = {
+                meta: { schema: {} },
+                create(context) {
+                    return {
+                        Program(node) {
+                            context.report({ node, message: "bad" });
+                        }
+                    };
+                }
+            };
 
-            assert.strictEqual(processStub.callCount, 1, "calls `process.emitWarning()` once");
-            assert.deepStrictEqual(
-                processStub.getCall(0).args,
-                [
-                    "\"rule-with-null-schema\" rule has options but is missing the \"meta.schema\" property and will stop working in ESLint v9. Please add a schema: https://eslint.org/docs/developer-guide/working-with-rules#options-schemas",
-                    "DeprecationWarning"
+            assert.throws(() => ruleTester.run("rule-with-empty-object-schema", ruleWithEmptyObjectSchema, {
+                valid: [],
+                invalid: [
+                    { code: "var foo = bar;", options: [{ foo: true }], errors: 1 }
                 ]
-            );
+            }),
+            "`schema: {}` is a no-op. For rules with options, please fill in a complete schema. For rules without options, please omit `schema` or use `schema: []`.");
         });
 
-        it("should not log a deprecation warning when schema is an empty array", () => {
+        it("should not throw an error when schema is an empty array", () => {
```

Comment:
Seems like these tests could be updated to remove the use of `should`.

Reply:
How else would you write it? I do want it to read like an English sentence.